### ISSUE TITLE
Importers

### ISSUE THREAD

**Issue by nabeelio** [flarum/core#1197]
Hi. Has any thought been given to how to write the importers? Any higher-level abstraction layer? An artisan command? I've got a forum on IPS with 12k+ topics and 80k replies that I want to move over. I don't mind taking a stab at an initial importer. Trying to find some schema docs. I wrote a few importers back in the day. It seems like y'all are using something of an abstraction with only small points of contact with Laravel, some pretty cool stuff!

**Comment by franzliedke** [comment]
Hi @nabeelio, thanks for your positive feedback! 😄 

I am sorry, I am not totally sure what this issue is about. Are you asking for an initial skeleton for our "official" importer tool? Or just some guidelines on what we are looking for so that someone (like you?) can get started building it?

**Comment by nabeelio** [comment]
@franzliedke Yes, both :) If a skeleton exists, or any plans for how. My thought was an artisan command with a config file or something

**Comment by franzliedke** [comment]
At the moment, we are still leaving this up to the community. There are several tools already for some of the existing forum software out there. We will likely take some of their work as inspiration for an official tool, once that comes around...

**Comment by nabeelio** [comment]
Cool... is there an up-to-date schema lying around somewhere? It'd be a good place to start.

**Comment by franzliedke** [comment]
Sorry for the slow response...

No, we don't have a schema lying around, I think? Your best bet is to install the latest version, and look at the database, I'm afraid.

**Comment by BartVB** [comment]
FYI this is the most elaborate migration script that I've found so far:
https://github.com/mondediefr/fluxbb_to_flarum/
by @mondediefr
But as far as I can tell this will break on large forums.
